### PR TITLE
Translation to Spanish for Workshop: Java Basics

### PR DESCRIPTION
**Pull request by apclouds**

### CODE REVIEW

**Comment by ozhang22** [comment]
@beagandica pleasehelp with the review!

File: content/espanol/java-basics/activity-0.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: "Usando Repl.It"
+date: 2020-07-11T00:00:00Z
+weight: 1
+draft: false
+---
+
+## Using Repl.it
```
Comment (beagandica) [inline content/espanol/java-basics/activity-0.md:8]:
```suggestion
## Usando Repl.it
```

File: content/espanol/java-basics/_index.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+description: "Una guía introductoría a conceptos básicos de programación en Java."
+date: 2020-07-07T00:00:00Z
+prereq: "Ninguno"
+difficulty: "Principiante"
+icon: "fab fa-java"
+draft: false
+hidden: false
+---
+
+## Overview
```
Comment (beagandica) [inline content/espanol/java-basics/_index.md:12]:
```suggestion
## Resumen general
```

File: content/espanol/java-basics/_index.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+<img src="images/logo.png" height="150"/> 
+
+Beinvenidos al curso <b>Java: Basics</b>, donde aprenderás un lenguaje comúnmente utilizado que le da instrucciones a una computadora para crear programas emocionantes!
```
Comment (beagandica) [inline content/espanol/java-basics/_index.md:16]:
```suggestion
Beinvenidos al curso de <b>Java: Basics</b>, donde aprenderás un lenguaje comúnmente utilizado, el cual le da instrucciones a una computadora para crear programas emocionantes!
```

File: content/espanol/java-basics/_index.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Java es un lenguage de programacíon simplificado creado por James Gosling junto a Sun Microsystems en California en el año 1995.
+
+En este curso, cubriremos las capacidades básicas del lenguaje.
+
+![animation of duck Patrick](https://media.giphy.com/media/l49JKwmJLChtS6d44/giphy.gif)
```
Comment (beagandica) [inline content/espanol/java-basics/_index.md:22]:
```suggestion
![animacion del pato Patricio](https://media.giphy.com/media/l49JKwmJLChtS6d44/giphy.gif) 
```

File: content/espanol/java-basics/_index.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+En este curso, cubriremos las capacidades básicas del lenguaje.
+
+![animation of duck Patrick](https://media.giphy.com/media/l49JKwmJLChtS6d44/giphy.gif) 
+
+<p style="text-align: center;">(image created by Molang: giphy.com/molangofficialpage)</p>
+
+En este curso, seguiremos el ejemplo de nuestro pato amarillo, amante a la música, llamado <b>Patrick</b> 🐥 para aprender todo sobre Java y hacer amistades nuevas en el camino.
+
+## Table of Contents
```
Comment (beagandica) [inline content/espanol/java-basics/_index.md:28]:
```suggestion
## Contenido
```

File: content/espanol/java-basics/activity-1.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Vamos a escribir un programa que diga `Hello World` luego de escribir la siguiente línea de código `System.out.print("Hello World");` 
+
+El programa imprimirá lo que escribas entre los paréntesis de `System.out.print(`. Cada línea de código termina con `;`.
+
+Añada más declaraciones para imprimir números, escribiendo los números entre los paréntesis (i.e. `System.out.print(2020)`) o cualquier símbolo, escribiendo el símbolo entre 2 comillas simples `'` (i.e. `System.out.print('@')`).
+
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsHelloWorld?lite=true#Main.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+{{% notice tip %}}
+#### Te fijaste que todas las delcaraciones que escribistes fueron impreses en una línea?
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:19]:
```suggestion
#### Te fijaste que todas las declaraciones que escribistes fueron impreses en una línea?
```

File: content/espanol/java-basics/activity-1.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Aveces queremos añadir una línea entre medio cuando escribimos. Para lograr eso, tenemos que usar el siguiente código `System.out.println();` 
+
+(Note: escribimos `println` en vez de `print` para crear una línea nueva)
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:22]:
```suggestion
(Nota: escribimos `println` en vez de `print` para crear una línea nueva)
```

File: content/espanol/java-basics/activity-1.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+{{% notice tip %}}
+#### Cómo imprimir multiples símbolos en una línea de código?
+Apredimos que para imprimir símbolos, los escribimos entre 2 comillas sencillas `'`, pero cómo podemos imprimir multiples símbolos en una línea?
+
+Para lograrlo, ponemos escribir los multiples símbolos entre 2 comillas dobles `"` (i.e. `System.out.print("@ , - h A #");`)
+
+{{% /notice %}}
+
+{{% notice tip %}}
+#### !! Importante !! Aveces no se puede escribir caracteres entre las comilas dobles " "
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:36]:
```suggestion
#### !! Importante !! Aveces no se puede escribir caracteres entre las comillas dobles " "
```

File: content/espanol/java-basics/activity-1.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+Cuando quieres poner caracteres entre medio de `" "` en una declaración de imprimir, es fácil confundir l computadora en cuanto a qué caracteres debería imprimir.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:37]:
```suggestion
Cuando quieres poner caracteres entre medio de `" "` en una declaración de imprimir, es fácil confundir a la computadora en cuanto a qué caracteres debería imprimir.
```

File: content/espanol/java-basics/activity-1.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+      / /)  )    \ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ /
+     /,--"-"- 
+
+
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsOwl?lite=true#Main.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+{{% notice warning %}}
+#### Tuve un error
+
+     Main.java:4: error: reached end of file while parsing
+Si llegas a un error como este, es posuble que se te haya olvidado <b>escape</b> caraceteres como `"` o `\`.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:66]:
```suggestion
Si llegas a un error como este, es posible que se te haya olvidado <b>escape</b> caraceteres como `"` o `\`.
```

File: content/espanol/java-basics/activity-1.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+Repasa el tercer tip para ver como <b>escape</b> caracteres en el método de print.
+{{% /notice %}}
+
+### 3. Make a Comment
+Cuando creamos un programa de Java, podemos dejar anotcaciones y comentarios en una línea añadiendo `//` o para multiples líneas usando `/*` y `*/` (Note: Las instrucciones en las actividades previas estaban escritas como comentarios). 
+
+Añadiendo comentarios no afecta so programa de ninguna manera. Puedes añadir cualquier cantidad de comentarios en tu programa.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-1.md:74]:
```suggestion
Añadir comentarios no afecta tu programa de ninguna manera. Puedes añadir cualquier cantidad de comentarios en tu programa.
```

File: content/espanol/java-basics/activity-2.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
+---
+title: "Variables and Types"
+description: "Introducción a types y variables en Java."
+date: 2020-07-08T00:00:00Z
+weight: 3
+---
+
+### 1. Variables: Define un Nombre
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:8]:
```suggestion
### 1. Variables: Define el nombre
```

File: content/espanol/java-basics/activity-2.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+En el último ejercisio, aprendimos como imprimir distintas delcaraciones con el método `System.out.print()`.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:9]:
```suggestion
En el último ejercicio, aprendimos como imprimir distintas delcaraciones con el método `System.out.print()`. 
```

File: content/espanol/java-basics/activity-2.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+date: 2020-07-08T00:00:00Z
+weight: 3
+---
+
+### 1. Variables: Define un Nombre
+En el último ejercisio, aprendimos como imprimir distintas delcaraciones con el método `System.out.print()`. 
+
+Está genial saber como imprimir números y oraciones. Pero todavía no les hemos dado un significado más profundo.
+
+En Java, ahora vamos a introducir el concepto de una <b>variable</b> for data. Una variable es como una caja donde guardas un pedazo de data o información con un nombre.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:13]:
```suggestion
En Java, ahora vamos a introducir el concepto de una <b>variable</b>. Una variable es como una caja donde guardas un pedazo de data o información con un nombre.
```

File: content/espanol/java-basics/activity-2.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+
+Por ejemplo, pueden haber una variable llamada  `myName` que guarda la información `"Patrick"`, y otra variable llamada `loveMusic` que guarda el valor de `true`, y otra variable llamada `age` que guarda el valor `3`. Como puedes ver en la siguiente imagen:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:15]:
```suggestion
Por ejemplo, puede haber una variable llamada  `myName` que guarda la información `"Patrick"`, y otra variable llamada `loveMusic` que guarda el valor de `true`, y otra variable llamada `age` que guarda el valor `3`. Como puedes ver en la siguiente imagen:
```

File: content/espanol/java-basics/activity-2.md [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+
+<img src="../images/variable.png" height="200"/> 
+
+### 2. Qué Type
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:19]:
```suggestion
### 2. Tipos de variables
```

File: content/espanol/java-basics/activity-2.md [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+
+### 2. Qué Type 
+
+Entonces, cómo podemos crear estas cajas o variables en Java para guardar data?
+
+Antes de aprender cómo, tenemos que aprender sobre el concepto de un <b>type</b> en Java. Cada variable tiene su propio type que especifica el tipo de data que se puede guardar en esa variable.
+
+Por ejemplo, la variable llamada `age` guarda el número `3` debería contener <b>integers</b>, y no oraciones ni símbolos. 
+
+Cuando usamos Java, para que una variable pueda guardar números enteros, la variable tiene que ser type <b>int</b>. Así podemos declarar una variable `int` llamada `age` con data `3` para guardar, con la siguient línea the código:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:27]:
```suggestion
Cuando usamos Java, para que una variable pueda guardar números enteros, la variable tiene que ser type <b>int</b>. Así podemos declarar una variable `int` llamada `age` con data `3` para guardar, con la siguiente línea the código:
```

File: content/espanol/java-basics/activity-2.md [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+```java
+int age = 3;
+```
+
+Por lo general, para declarar una variable, se utiliza el siguiente formato: `[data type] [variable name] = [data];`.
+{{% notice note %}}
+Los siguientes son distintos tipos de variables ya disponible en Java:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:34]:
```suggestion
Los siguientes son distintos tipos de variables ya disponibles en Java:
```

File: content/espanol/java-basics/activity-2.md [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+
+- `int` type: un número entero (i.e. `20`, `30`, `35`).
+
+- `char` type: un caracter sea símbolo or letra individual del abecedario (i.e. `'A'`, `'b'`, `'('`, `']'`).
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:38]:
```suggestion
- `char` type: un caracter, ya sea símbolo or letra individual del abecedario (i.e. `'A'`, `'b'`, `'('`, `']'`). 
```

File: content/espanol/java-basics/activity-2.md [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+<img src="../images/dataType.png" height="200"/> 
+
+Para declarar estas variables en Java, escribimos lo siguiente:
+```java
+String name = "Patrick";
+int age = 10;
+boolean loveMusic = true;
+```
+
+### 3. Cuál es el Type -- Ayuda a Patrick
```
Comment (beagandica) [inline content/espanol/java-basics/activity-2.md:58]:
```suggestion
### 3. Cuál es el Type o Tipo de variable -- Ayuda a Patrick
```

File: content/espanol/java-basics/activity-3.md [hunk 21]
```diff
@@ -0,0 +1,107 @@
+---
+title: "Operadores"
+description: "Introduccíon a operadores y cómo realizar operaciones a variables en Java."
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:3]:
```suggestion
description: "Introduccíon de operadores y cómo realizar operaciones a variables en Java."
```

File: content/espanol/java-basics/activity-3.md [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+date: 2020-07-13T00:00:00Z
+weight: 4
+---
+
+### 1. Matemática : Operadores de Artimétic
+En el último ejercisio, aprendimos sobre distintos tipos de variables y cómo podemos aignarles un valor. Por ejemplo:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:9]:
```suggestion
En el último ejercicio, aprendimos sobre distintos tipos de variables y cómo podemos aignarles un valor. Por ejemplo:
```

File: content/espanol/java-basics/activity-3.md [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+
+```java
+int age = 3;  // declara la variable age de type int con valor de 3
+age = 4;      // asingnale el valor de 4 a la variable age
+```
+
+El código de arriba le suma 1 a la variable `age`. 
+
+Y en Java, podmeos realizar operaciones matemáticas directamente con las variables y los siguientes <b>opereadores aritméticos</b>: `+`, `-`, `*`, `/`, `%`.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:18]:
```suggestion
Y en Java, podemos realizar operaciones matemáticas directamente en las variables y con los siguientes <b>opereadores aritméticos</b>: `+`, `-`, `*`, `/`, `%`.
```

File: content/espanol/java-basics/activity-3.md [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+Por ejemplo, podemos usar el operator de suma `+` para sumarle 1 a la variable `age`:
+```java
+int age = 3;  // declara la variable age de type int con valor de 3
+age = age + 1;// suma 1 a la variable age
+```
+{{% notice note %}}
+List de operadores de aritmética en Java:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:25]:
```suggestion
Lista de operadores de aritmética en Java:
```

File: content/espanol/java-basics/activity-3.md [hunk 25]
```diff
@@ -0,0 +1,107 @@
...
+- `<`: menor a.
+
+- `>=`: mayor a o igual a.
+
+- `<=`: menor a o igual a.
+
+Note: puedes usar  `==` y `!=` en dos variables con data del mismo tipo.
+{{% /notice %}}
+
+### 3. Estadística de examen
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:67]:
```suggestion
### 3. Examen de estadística
```

File: content/espanol/java-basics/activity-3.md [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+
+- `<=`: menor a o igual a.
+
+Note: puedes usar  `==` y `!=` en dos variables con data del mismo tipo.
+{{% /notice %}}
+
+### 3. Estadística de examen 
+Patrick 🐥 y 4 alumnos acaban de tener un examen de música. Los resultados de sus exámenes: Patrick(88), Tom(89), Mary(95), Chris(84), Jen(92).
+
+Queremos producir un reporte para estádisticas de examen de la clase de música.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-3.md:70]:
```suggestion
Queremos producir un reporte de estádisticas de la clase de música. 
```

File: content/espanol/java-basics/activity-7.md [hunk 27]
```diff
@@ -0,0 +1,138 @@
...
+(Hint: Utilize for loop y `size()`).
+
+Luego de `Run`, el programa debe imprimir lo siguiente:
+```
+Birds Of Prey, The Birds, Angry Birds, Bird Box, The Secret Life of Pets
+```
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsList?lite=true#Bird.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+### Fin de Workshop 🐥 
+<b>Patrick</b> 🐥 gracias por participar. Esperamos que haya aprendido mucho con este curso :)
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:135]:
```suggestion
<b>Patrick</b> 🐥 gracias por participar. Esperamos que hayas aprendido mucho en este curso :)
```

File: content/espanol/java-basics/activity-7.md [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+#### Ejercisio: Escriba un método `printList()` que tome un ArrayList como entrada e imprime todos los elementos del ArryList en orden 
+(Hint: Utilize for loop y `size()`).
+
+Luego de `Run`, el programa debe imprimir lo siguiente:
+```
+Birds Of Prey, The Birds, Angry Birds, Bird Box, The Secret Life of Pets
+```
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsList?lite=true#Bird.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+### Fin de Workshop 🐥
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:134]:
```suggestion
### Fin del Workshop 🐥 
```

File: content/espanol/java-basics/activity-7.md [hunk 29]
```diff
@@ -0,0 +1,138 @@
...
+Para remover un elemento de un ArrayList, llamamos el método `remove()`. 
+Por ejemplo:
+```java
+System.out.print(topMoviesList.size()); // imprime 5
+topMoviesList.remove("Birds Of Prey");
+System.out.print(topMoviesList.size()); // imprime 4
+```
+<br />
+
+#### Ejercisio: Escriba un método `printList()` que tome un ArrayList como entrada e imprime todos los elementos del ArryList en orden
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:125]:
```suggestion
#### Ejercicio: Escriba un método `printList()` que tome un ArrayList como entrada e imprime todos los elementos del ArryList en orden 
```

File: content/espanol/java-basics/activity-7.md [hunk 30]
```diff
@@ -0,0 +1,138 @@
...
+
+Es útil saber cuantos espaciosa habrán en un array luego de crearse (por ejemplo, para accesar el último espacio en un array). También podemos conseguir el número de espacios en un array con `length` al escribir `array_name.length`. 
+
+Por ejemplo:
+```java
+int numOfMovie = topMovies.length; // numOfMovie tiene un valor de 5
+```
+<br />
+
+#### Ejercisio: Esriber un método `printArr()` que tome un array como entrada e imprima todos los elementos en el array en orden
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:69]:
```suggestion
#### Ejercicio: Esriber un método `printArr()` que tome un array como entrada e imprima todos los elementos en el array en orden 
```

File: content/espanol/java-basics/activity-7.md [hunk 31]
```diff
@@ -0,0 +1,138 @@
...
+```
+Al giaul que podemos accesar la data con un índice, también podemos accesar y modificar cada pedazo de data con su índice correspondiente. 
+
+Por ejemplo:
+```java
+String movie1 = topMovies[0]; // movie1 tiene el valor de "Birds Of Prey"
+```
+{{% /notice %}}
+
+Es útil saber cuantos espaciosa habrán en un array luego de crearse (por ejemplo, para accesar el último espacio en un array). También podemos conseguir el número de espacios en un array con `length` al escribir `array_name.length`.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:61]:
```suggestion
Es útil saber cuantos espacios habrán en un array luego de crearse (por ejemplo, para accesar el último espacio en un array). También podemos conseguir el número de espacios en un array con `length` al escribir `array_name.length`. 
```

File: content/espanol/java-basics/activity-7.md [hunk 32]
```diff
@@ -0,0 +1,138 @@
...
+
+Sin embargo, cuando la computadora lee `arr[1]`, no piensa en ver el `1er` espacio de `arr`. 
+
+En realidad, de la manera que la computadora lo piensa es `arr[1]` significa ver el espacio a `1` distancia del principo del array.
+
+En otras palabras, el índice de hecho es desplazamiento de memoria. Para accesar el primer elemento debemos conseguir el elemento que está a una distancia de `0` desde el principio del array (i.e. `arr[0]`).
+{{% /notice %}}
+
+{{% notice tip %}}
+#### Una alternativa para crear un array:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:46]:
```suggestion
#### Alternativa para crear un array:
```

File: content/espanol/java-basics/activity-7.md [hunk 33]
```diff
@@ -0,0 +1,138 @@
...
+```java
+topMovies[0] = "Birds Of Prey";
+topMovies[1] = "The Birds";
+topMovies[2] = "Angry Birds";
+topMovies[3] = "Bird Box";
+topMovies[4] = "The Secret Life of Pets";
+// el array topMovies ahora guarda "Birds Of Prey", "The Birds", "Angry Birds", "Bird Box", "The Secret Life of Pets" en ese orden
+```
+{{% notice info %}}
+#### Por qué el índice de un array cominezo en 0?
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:35]:
```suggestion
#### Por qué el índice de un array comineza en 0?
```

File: content/espanol/java-basics/activity-7.md [hunk 34]
```diff
@@ -0,0 +1,138 @@
+---
+title: "Data Structures (Array & ArrayList)"
+description: "Introduce basic data structures in Java."
+date: 2020-07-16T00:00:00Z
+weight: 8
+---
+En los últimos ejercisios, aprendimos sobre varilables que nos permiten accesar y guardar un pedazo de data. Pero, y si quereos guardar mucha data parecida? Habrá una manera buena de organizarlas sin declarar multiples variables?
```
Comment (beagandica) [inline content/espanol/java-basics/activity-7.md:7]:
```suggestion
En los últimos ejercicios, aprendimos sobre varilables que nos permiten accesar y guardar un pedazo de data. Pero, y si quereos guardar mucha data parecida? Habrá una manera buena de organizarlas sin declarar multiples variables?
```

File: content/espanol/java-basics/activity-4.md [hunk 35]
```diff
@@ -0,0 +1,133 @@
...
+#### Ejemplo
+```java
+boolean likeMusic = true;
+if(likeMusic == true){
+    System.out.println("I like Music");
+}else{
+    System.out.println("I don't like Music");
+}
+```
+El código imprimirá `I like Music` porque la expresión boolean, `likeMusic == true`, es igual al valor de true.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-4.md:54]:
```suggestion
El código imprimirá `I like Music` (o 'me gusta la musica') porque la expresión boolean, `likeMusic == true`, es igual al valor de true.
```

File: content/espanol/java-basics/activity-4.md [hunk 36]
```diff
@@ -0,0 +1,133 @@
...
+Patrick debería decir "Good Morning" entre 0:00 - 11:59, "Good Afternoon" entre 12:00 - 19:59, y "Good Night" entre 20:00 - 23:59.
+
+1. Tenemos la variable `currentHour` que guarda la hora actual. 
+2. Llene la correcta `boolean expression A` y `boolean expression B` para completar su programa. 
+3. Pruebe su resultado asignando la variable `currentHour` a un número entre 0 y 23:
+
+<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/JavaBasicsIfStatement?lite=true#Main.java" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+### 2. Repítalo de nuevo: For-Loop, While-Loop
+En clase de Matemáticas, la maestra de Patrick 🐥 le pregunta a la clase que solucione un problema, y tal vez Patrick necesite ayuda:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-4.md:68]:
```suggestion
En clase de matemáticas, la maestra de Patrick 🐥 le pregunta a la clase que solucione un problema, y tal vez Patrick necesite ayuda:
```

File: content/espanol/java-basics/activity-4.md [hunk 37]
```diff
@@ -0,0 +1,133 @@
...
+
+Lo siguiente es un ejemplo que imprime los números entre el 1 y el 10:
+```java
+for(int number = 1; number <= 10; number = number + 1){
+    System.out.println(number);
+}
+```
+{{% /notice %}}
+
+Luego de aprender sobre el for loop y el while loop, vamos a ver si Patrick 🐥 ouede calcular calculate <b>1 + 2 + 3 + 4 ... + 99 + 100</b> fácilmente?
```
Comment (beagandica) [inline content/espanol/java-basics/activity-4.md:114]:
```suggestion
Luego de aprender sobre el for loop y el while loop, vamos a ver si Patrick 🐥 puede calcular <b>1 + 2 + 3 + 4 ... + 99 + 100</b> fácilmente?
```

File: content/espanol/java-basics/activity-5.md [hunk 38]
```diff
@@ -0,0 +1,100 @@
+---
+title: "Métodos"
+description: "Introducción a métodos en Java."
+date: 2020-07-13T00:00:00Z
+weight: 6
+---
+
+### 1. Qué Son Métodos?
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:8]:
```suggestion
### 1. Qué son los métodos?
```

File: content/espanol/java-basics/activity-5.md [hunk 39]
```diff
@@ -0,0 +1,100 @@
...
+int num = 1;
+while(num <= 100){
+    total = total + num;
+    num = num + 1;
+}
+System.out.println("Sum: " + total);
+```
+Pero si en vez queremos calcular la suma del 1 al 77? Cómo podemos lograr eso sin tener que reescribir el mismo código?
+
+Queremos crear un ✨ magic box ✨ en el código sin importar para qué dos números queremos para la suma.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:23]:
```suggestion
Queremos crear un ✨ magic box ✨ ( o caja magica) en el código sin importar los dos números que queremos sumar.
```

File: content/espanol/java-basics/activity-5.md [hunk 40]
```diff
@@ -0,0 +1,100 @@
...
+<img src="../images/method.png" height="250"/> 
+
+En Java, un <b>method</b> es como un magic box que ejecuta ciertos pasos al ejecutar un código usando los valores que ingrese un usuario.
+
+Hay 2 partes en un method: header y body:
+
+{{% notice note %}}
+#### Header de un Método
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:31]:
```suggestion
#### Header (o cabecero) de un Método
```

File: content/espanol/java-basics/activity-5.md [hunk 41]
```diff
@@ -0,0 +1,100 @@
...
+### 2. Cuenta la Pirámide!🔺
+<img src="../images/pyramid.png" height="250"/> 
+<p style="text-align: center;">(photo credit: aliexpress.com/item/32306945847.html)</p>
+
+La imagen de arriba es de una pirámide de números cuadrados donde cada nivel es una raíz cuadrada perfecta del número del nivel actual desde el tope.
+
+Es decir, el nivel del tope tiene `1 * 1` beads, el segundo nivel tiene `2 * 2` beads.
+
+<br />
+Vamos a escribir un método que toma el número total de niveles y resulta el número total de beads in la pirámide!
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:93]:
```suggestion
Vamos a escribir un método que toma el número total de niveles y resulta el número total de beads ( o pepitas) en la pirámide!
```

File: content/espanol/java-basics/activity-5.md [hunk 42]
```diff
@@ -0,0 +1,100 @@
...
+
+Luego de finalizar el ejercisio, verás el siguiente mensaje:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:95]:
```suggestion
Luego de finalizar el ejercicio, verás el siguiente mensaje:
```

File: content/espanol/java-basics/activity-5.md [hunk 43]
```diff
@@ -0,0 +1,100 @@
...
+```
+Congratulations! Challenge Solved!
```
Comment (beagandica) [inline content/espanol/java-basics/activity-5.md:97]:
```suggestion
Felicidades! Haz resuelto el desafío!
```

File: content/espanol/java-basics/activity-6.md [hunk 44]
```diff
@@ -0,0 +1,183 @@
...
+Por ejemplo, la línea de código que usamos en la actividad 1 es una llamada de un método de una clase pre definida llamada `System`.
+```java
+System.out.println("Hello World");
+```
+1. `System` es una clase que se usa para manejar entradas y salidas del usuario.
+2. `out` es una variable en la clase `System` de clas/tipo `PrintStream`.
+3. `println()` es un método pre definido en la clase `PrintStream`.
+
+<br />
+Otra clase con la cual hemos interactuados en ejercisios previos es `String`. La clase `String` defines un par de reglas que determinan como una lista de caracteres deberían comportarse.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:20]:
```suggestion
Otra clase con la cual hemos interactuados en ejercicios previos es `String`. La clase `String` define un par de reglas que determinan como una lista de caracteres deberían comportarse.
```

File: content/espanol/java-basics/activity-6.md [hunk 45]
```diff
@@ -0,0 +1,183 @@
...
+#### Por último, vamos a crear unos métodos para la clase `Bird`.
+Vamos a crear 6 métodos para esta clase.
+-  getSpecies();   // retorna la especie de un pájaro
+-  getName();      // retorna el nombre de un pájaro
+-  getHobby();     // retorna el pasa tiempo de un pájaro
+-  getAge();       // retorna la edad de un pájaro
+-  getLoveMusic(); // retorna si al pájaro ama la música
+-  toString();     // retorna la information como un string
+
+Intente escribir los primeros 5 métodos con lo que ha aprendido en los ejercisios previos.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:113]:
```suggestion
Intente escribir los primeros 5 métodos con lo que ha aprendido en los ejercicios previos.
```

File: content/espanol/java-basics/activity-6.md [hunk 46]
```diff
@@ -0,0 +1,183 @@
...
+    }else{        
+        info = info + "However, I do not like music!";
+    }
+
+    // retorna el valor de la variable info
+    return info;
+}
+```
+{{% notice warning %}}
+Si implementaste corectamente la clase`Bird`, verás el siguiente mensaje luego de hacer click a `Run`:
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:144]:
```suggestion
Si implementaste corectamente la clase `Bird`, verás el siguiente mensaje luego de hacer click en `Run`:
```

File: content/espanol/java-basics/activity-6.md [hunk 47]
```diff
@@ -0,0 +1,183 @@
...
+```
+Congratulations! You correctly implemented the Bird Class :)
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:146]:
```suggestion
Felicidades! Haz implementado la clase Bird de forma correcta :)
```

File: content/espanol/java-basics/activity-6.md [hunk 48]
```diff
@@ -0,0 +1,183 @@
...
+```
+
+Si hay algo incompleto o incorrecto cuando oprimas `Run`:
+- Tal vez te salga el error `Main.java:3: error: cannot access Bird`.
+- O si no, el mensaje, `Something is still not quite right!`.
+{{% /notice %}}
+
+### 3. Creando Pájaros 🐥
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:154]:
```suggestion
### 3. Usando la clase Bird 🐥
```

File: content/espanol/java-basics/activity-6.md [hunk 49]
```diff
@@ -0,0 +1,183 @@
...
+Felicitaciones! Has escrito su primera clase en Java. `Bird` class. Ahora vamos a escribir un programa que utilice la clase para crea objectos de tipo `Bird`.
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:155]:
```suggestion
Felicitaciones! Haz escrito su primera clase en Java, `Bird` class. Ahora vamos a escribir un programa que utilice la clase para crea objectos de tipo `Bird`.
```

File: content/espanol/java-basics/activity-6.md [hunk 50]
```diff
@@ -0,0 +1,183 @@
...
+
+Definimos la clase `Bird` para tener los siguientes atributos (variables de instancia) and y comportamientos (métodos):
+<img src="../images/bird_class.png" height="450"/> 
+
+
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:160]:
```suggestion
```

File: content/espanol/java-basics/activity-6.md [hunk 51]
```diff
@@ -0,0 +1,183 @@
...
+```
+Intenta crear multiples pájaros de distintas especies usando la clase Bird 🐦🐤🐔🐧
+
+<br />
+Próximo, usaremos el método `toString()` en los objetos de tipo `Bird` para imprimir la información de nuestros amigos. Puedes hacer una de las siguientes:
+
+1. Imprimir el valor que retorna el método `toString()`. (i.e. `System.out.println(patrick.toString());`)
+2. Imprimir el objecto `Bird` quen en cambio llama el método `toString()` (i.e. `System.out.println(patrick);`).
+
+Intante imprimir la informacíon de todos sus amigos pájaros 🐦🐤🐔🐧
```
Comment (beagandica) [inline content/espanol/java-basics/activity-6.md:180]:
```suggestion
Intenta imprimir la informacíon de todos sus amigos pájaros 🐦🐤🐔🐧
```

**Review by ozhang22 — APPROVED** [review]
Approved.

**Pull request merged**